ppd/ppd-test.c

```diff
@@ -160,20 +162,20 @@ cups_array_t *ppdTest(int ignore,          // Which errors to ignore
   if (q_with_v == 1)
   {
   
-    ppdArrayAdd(output,
-                        "testppdfile: The -q option is incompatible with the -v option.");
+    cupsArrayAdd(output,
```

Comment:
You do not need to mention these string constants twice here. If you define an appropriate log function in the `testppdfile.c` file and call `ppdTest()` with this log function, `ppdTest()` does not need to create an array for the human-readable output at all, but simply only log.

The log function does not even need to create an output array then, but can simply `printf()` the log messages.

Reply:
I have created the output array to solve the problem of when the utility is to be run from a .c file and its output is to be displayed in a different space, for e.g. in the web utility of cups and not in terminal, the output proves to be helpful as one can get the output array and use it to display the results.